crates/torii/core/src/engine.rs

```diff
@@ -303,26 +307,26 @@ impl<P: Provider + Sync> Engine<P> {
                     }
                 }
 
-                if let Some((_, last_tx_hash)) = transactions.last() {
-                    // Dedup transactions
-                    // As me might have multiple events for the same transaction
-                    if *last_tx_hash == event.transaction_hash {
-                        continue;
-                    }
-                }
-                transactions.push((block_number, event.transaction_hash));
+                transactions
+                    .entry((block_number, event.transaction_hash))
+                    .or_insert(vec![])
+                    .push(event);
             }
         }
 
+        debug!("Transactions: {}", &transactions.len());
```

Comment:
Should we leave these debugs?

Reply:
this are tracing::debug so its okay to keep them